sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -253,7 +270,8 @@ case class CatalogTable(
     tracksPartitionsInCatalog: Boolean = false,
     schemaPreservesCase: Boolean = true,
     ignoredProperties: Map[String, String] = Map.empty,
-    viewOriginalText: Option[String] = None) {
+    viewOriginalText: Option[String] = None,
+    clusterBySpec: Option[ClusterBySpec] = None) {
```

Comment:
We haven't added any field to `CatalogTable` in the last 5 years and should we _not_ add this new field? Alternatively, we can store clustering columns in the property and make the property as reserved?

Reply:
Since we have to store it in the table properties anyway at the Hive layer, I think it's simpler to just do it ahead here.

Reply:
Done, I guess no need to make it as a reserved property like properties in `TableCatalog`?